lgn-provers/src/provers/v1/groth16/euclid_prover.rs

```diff
@@ -36,10 +35,8 @@ impl Groth16Prover {
         debug!("Groth16 prover created");
         Ok(Self { inner })
     }
-}
 
-impl Prover for Groth16Prover {
-    fn prove(
+    pub(super) fn prove(
         &self,
         revelation: &[u8],
     ) -> Result<Vec<u8>> {
```

Comment:
I didn't want to move the code around to make reviewing easier. This makes the methods available for the `task.rs` so it can call it, now that the traits are gone. This pattern repeats on all provers.